server/src/main/java/org/elasticsearch/plugins/internal/document_parsing_observer/DocumentParsingObserverFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.internal.document_parsing_observer;
+
+/**
+ * An interface of a factory to create a DocumentParsingObserver.
+ * A default implementation is returning a noop DocumentParsingObserver
+ */
+public interface DocumentParsingObserverFactory {
```

Comment:
Could we just use `Supplier<DocumentParsingObserver` rather than a separate type?

Reply:
will do